text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+- The Resource Coordinator will detect conflicts in Entity of the same type being discovered and choose one to use.
+- When using Entity Detectors and Resource detectors together, the following merge rules will be used:
+  - Entity merging will occur first resulting in an "Entity Merged" Resource.
+    - Entities of different types will be merged into the resulting Resource.
+    - Entities of the same type will have one rejected and one accepted, based on priority.
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the rules around merging Resource across schema-url will be dropped.  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it.  Additionally, as no Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent violation of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
+  - An OOTB "Env Variable Entity Detector" will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Coordinator would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK.  For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod.   SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
+
+#### Entity Merging and Resource
+
+The most important aspect of this design is how Entities will be merged to construct a Resource. We provide a simple algorithm for this behavior:
+
+- Construct a set of detected entities, E
+- All entity detectors are sorted by priority
+- For each entity detector
+  - For each entity detected
+    - If the entity exists in E, ignore it
+    - Otherwise, add the entity to E
+- Construct a Resource from the set E.
+
+Any implementation that achieves the same result as this algorithm is acceptable.
```

Comment:
What do you mean by this sentence? Do you imagine efficiency problems with this that would suggest alternative implementations? What is accomplished by leaving the door open to alternatives?

Reply:
The algorithm as defined is very imperative/top down.  I can imagine bottom up implementations or others.  I don't think we gain anything by forcing an algorithm, and I do expect folks could improve upon this.